[Owen Tan] iP

mslevis opened the pull request:
# Duke 
> “Productivity is being able to do things that you were never able to do before.” – Franz Kafka ([source](https://dansilvestre.com/productivity-quotes/))

Have a busy schedule? Duke frees your mind from your mundane tasks in life. It's,

* text-based
* easy to learn
* minimalistic
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](www.google.com).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 

And it is absolutely **FREE!** :moneybag:

Features: 
- [x] Managing tasks
- [x] Managing deadlines
- [ ] GUI (coming soon)
- [ ] Reminders (coming soon)

---
Here is some code for you to practise Java.
```java
if (isAwesome) {
    return true
}
```

Commit:
Add Gradle support

mslevis committed:
Added greeting, echo, exit

mslevis committed:
Include Add, List into Duke.java

mslevis committed:
Add Task class

mslevis committed:
Add ability to mark tasks as done

mslevis committed:
Abstract completing a task into a method

mslevis committed:
Add Todo, Deadline, Event

mslevis committed:
Add Text UI Testing

mslevis committed:
Add Exceptions Handling

mslevis committed:
Add delete function, Tidy up exception handling

mslevis committed:
Edit formatting, Improve UI Testing

Commit:
updated constructors for Tasks

mslevis committed:
Todo, Event, Deadline: Include PrintFormat method

mslevis committed:
Create Storage class

mslevis committed:
Add LocalDateTime type to Deadline, Event

mslevis committed:
Add Date parsing in Duke class

mslevis committed:
Change date output format

mslevis committed:
Remove imports

mslevis committed:
Merge branch 'branch-Level-7'

mslevis committed:
Merge conflicts between master and branch-Level-8

mslevis committed:
fix bugs

mslevis committed:
Create Classes Parser, TaskList, Ui

mslevis committed:
Add package

mslevis committed:
Add Junit Tests

mslevis committed:
Add Tests

mslevis committed:
Add Parser Tests

mslevis committed:
Update .gitignore

mslevis committed:
Remove data files

mslevis committed:
Change enum to match coding standard

mslevis committed:
Update Commands Package to match coding standard

mslevis committed:
Change to match coding standard

mslevis committed:
Implement search function

mslevis committed:
Add Javadocs for Task

mslevis committed:
Merge branch 'branch-Level-9'

mslevis committed:
Merge branch-A-CodingStandard

mslevis committed:
Merge branch-A-JavaDoc

mslevis committed:
Change version number

kslui99 commented on src/main/java/duke/commands/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    @Override
+    public String printFormat() {
```
Should this method be renamed to something like toPrintFormat(), so that it sounds more like a verb? The same applies to the printFormat() method in the Event, Task and Todo classes.

kslui99 commented on src/main/java/duke/commands/Task.java:
```diff
@@ -0,0 +1,41 @@
+        return getStatusIcon() + " " + description;
+    }
+
+    public boolean findKeyword(String keyword) {
```
Should this be renamed to something along the lines of canFindKeyword, so that it sounds more like a boolean?

kslui99 commented on src/main/java/duke/data/TaskList.java:
```diff
@@ -0,0 +1,76 @@
+        lst = new ArrayList<Task>();
+    }
+
+    public void iterList() {
```
Should this be renamed to something along the lines of printTaskList(), so that it more accurately describes what the function does? Because this function actually prints to the screen, instead of just iterating through the list.

kslui99 commented on src/main/java/duke/data/TaskList.java:
```diff
@@ -0,0 +1,76 @@
+        return lst;
+    }
+
+    public int count() {
```
Should this be renamed to getCount() to sound more like a verb?

kslui99 commented on src/main/java/duke/data/TaskList.java:
```diff
@@ -0,0 +1,76 @@
+        return task;
+    }
+
+    public void searchKeyword(String keyword) {
```
Should this be renamed to printMatchingTasks() so that it more accurately describes what the function does? Because the function actually prints the matching tasks to the screen.

kslui99 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,91 @@
+    private Task parseTask(String task) {
+        String[] tokens = task.split(" \\| ");
+        Task t = null;
+        // T | 0 | description | addInfo
```
Did you unintentionally leave this comment here in your submission?

kslui99 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,39 @@
+    public Ui(TaskList tasks) {
+        this.tasks = tasks;
+    }
+    public void greeting() {
```
Should this be renamed to printGreeting() so that it more accurately describes the function, since the function prints the greeting to the screen? Renaming it this way would make it sound more like a verb as well.

The same goes for the other methods below in this class.

kslui99 reviewed (commented):
Overall, I think you did a great job!

However, perhaps you could add more Javadoc comments as you code along the way for all classes and public methods.

mslevis committed:
Improve JavaDocs

mslevis committed:
Improve JavaDocs

NoraYUuu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,55 @@
+import duke.data.TaskList;
```
Seems no violation of coding standard in this file. Nice!

NoraYUuu commented on src/main/java/duke/commands/Task.java:
```diff
@@ -0,0 +1,68 @@
+     * @return A string representation of Task for saving.
+     */
+    public abstract String printFormat();
+}
```
Also seems no violation of coding standard!

NoraYUuu replied:
Except for the separation between description and param section. I noticed that many other methods do have this separation, maybe this absence of separation is just from previous commits, then it's fine.

NoraYUuu commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,139 @@
+        String[] tokens = cmd.split(" ");
+        Keyword command = validateCommand(tokens);
+
+        int middle;
+        int index;
```
Maybe can initialize variables here to void it being referred to while still being uninitialized?

NoraYUuu commented on src/main/java/duke/commands/Deadline.java:
```diff
@@ -0,0 +1,58 @@
+    protected LocalDateTime by;
+
+    /**
+     * Constructor for Deadline.
```
Would it be clearer to separate description and params by an empty line?

NoraYUuu replied:
This modification can be considered for several other places as well.

mslevis replied:
Point taken! Thank you.

NoraYUuu reviewed (commented):
I could hardly find violations. Good job!

mslevis committed:
Improve code quality

mslevis committed:
Merge branch 'add-gradle-support' of github.com:mslevis/ip into A-Gradle

mslevis committed:
Add Gradle Support

mslevis committed:
Add Gradle support

mslevis committed:
Merge branch A-Gradle

mslevis committed:
Change import checkstyle

mslevis committed:
Merge branch 'branch-A-CheckStyle'

mslevis committed:
Add JavaFX to Duke

mslevis committed:
Fix MainWindow.fxml

mslevis committed:
Change images

mslevis committed:
Modify Ui to return string

mslevis committed:
Integrate Duke functions into GUI

mslevis committed:
Merge branch 'branch-Level-10'

Li-Lehao commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,75 @@
+        new Duke("data/tasks.txt").run();
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
```
Maybe you can replace this default function header with a more meaningful one when you have time?
Add a @return tag and simply describe the getResponse() method.

Li-Lehao commented on src/main/java/duke/commands/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+        this.by = by;
+    }
+
+    /**
+     * Returns a string representation of Deadline.
+     * @return A string representation of Deadline.
+     */
```
I am not sure whether we need an empty line between Returns... and @return. I remember in cs2030 we need that empty line. Maybe we need to double-check that.

Li-Lehao commented on src/main/java/duke/controller/MainWindow.java:
```diff
@@ -0,0 +1,55 @@
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/todo.jpeg"));
+
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+        dialogContainer.getChildren().add(DialogBox.getDukeDialog(Ui.showGreeting(), dukeImage));
```
These two lines seem a bit too long and not easy to read. Maybe you can consider separate them? 
Overall you did a really good job! Almost cannot find any coding standard violation in your project!

mslevis committed:
Include Java assertions

Assumptions are made when writing parts of the code.

Let's use java assertions to help avoid crashes due to developer errors.

mslevis committed:
Improve code quality

mslevis committed:
Merge pull request #2 from mslevis/branch-A-CodeQuality

Improve code quality

mslevis committed:
Merge pull request #1 from mslevis/branch-A-Assertions

Include Java assertions

mslevis committed:
Merge branch 'master' into branch-A-Assertions

mslevis committed:
Set up CI

mslevis committed:
Add Circular DP

mslevis committed:
Add Bye Functionality

mslevis committed:
Modify circle of DP

mslevis committed:
Add comments

mslevis committed:
Handle empty input

mslevis committed:
Add notes field to Tasks

mslevis committed:
Modify Parser to fit notes

mslevis committed:
Update Storage methods to include notes

mslevis committed:
Merge branch 'branch-D-Notes'

mslevis committed:
Remove some images

mslevis committed:
Assert Parse format

mslevis committed:
Merge assertion to master

mslevis committed:
Update test cases

mslevis committed:
Remove I/O redirection tests from workflow

mslevis committed:
Improve GUI

mslevis committed:
Update GUI

Li-Lehao commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,167 @@
+     * @param cmd String from user's input.
+     * @throws DukeException
+     */
+    public String parse(String cmd) throws DukeException {
```
This parse() method is a bit too long and there might be some violation of SLAP rule. Maybe we can consider extracting some String calculation, task initialization and task execution to form new methods for better abstraction.
Other than that I think you have done a really good job. The code is clean and neat.

mslevis committed:
Fix height of Label

mslevis committed:
Rename Duke to Aoi

mslevis committed:
Add Icon

mslevis committed:
Remove redundant images

mslevis committed:
Add Ui screenshot

mslevis committed:
Set theme jekyll-theme-architect

mslevis committed:
Set theme jekyll-theme-architect

mslevis committed:
Add UserGuide

mslevis committed:
Merge branch 'master' of github.com:mslevis/ip

mslevis committed:
Set theme jekyll-theme-minimal

mslevis committed:
Update README

mslevis committed:
Create Command Class to encapsulate the execution of commands

There can be further abstraction of the execution of commands from
user's input.

Creating a Command class helps to abstract away the details of executing
the commands.

Lets,
* extract those executions into their own methods
* work on improving code quality by adhering to SLAP when writing the
  methods

mslevis committed:
Add Commands Class

mslevis committed:
Improve GUI

mslevis committed:
Add support for friendlier syntax commands

mslevis committed:
Add Help

mslevis committed:
Improve GUI

mslevis committed:
Update UI.png

mslevis committed:
Update README

mslevis committed:
Add badges to README

mslevis committed:
Add badges to README
